luarules/gadgets/unit_projectile_leash.lua

```diff
@@ -0,0 +1,181 @@
+function gadget:GetInfo()
+	return {
+		name = "Projectile Over-Range and Leashing",
+		desc = "Destroys projectiles if they exceed defined ranges",
+		author = "SethDGamre",
+		layer = 1,
+		enabled = true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+---- unit customParams ----
```

Comment:
consider choosing shorter names like overrange_distance, leash_range, maybe smth for projectile_destruction_method too, but nothing comes to mind.